cranelift-codegen/meta/src/isa/x86/legalize.rs

```diff
@@ -290,4 +301,84 @@ pub fn define(shared: &mut SharedDefinitions, x86_instructions: &InstructionGrou
     );
 
     group.build_and_add_to(&mut shared.transform_groups);
+
+    let mut narrow = TransformGroupBuilder::new(
+        "x86_narrow",
+        r#"
+    Legalize instructions by narrowing.
+
+    Use x86-specific instructions if needed."#,
+    )
+    .isa("x86")
+    .chain_with(shared.transform_groups.by_name("narrow").id);
+
+    // SIMD
+    let broadcast_immediate_u8 = Literal::constant(uimm8, 0x00);
+    let broadcast_immediate_f64 = Literal::constant(ieee64, 0x00);
```

Comment:
`let ieee64_0x00`?

Reply:
No, this is a good find. I'm pretty sure I actually should be using `f64::from_bits` since the whole point of this is to fill up an XMM register with zeroes (there may be a better way to do this but I saw this approach already emitted the XORPS I needed at https://github.com/CraneStation/cranelift/blob/master/cranelift-codegen/meta/src/isa/x86/encodings.rs#L1421-L1438). Let me fix the `From` below.